### ISSUE TITLE
Emphasize heading titles in messages

### ISSUE THREAD

**Issue by neiljp** [zulip/zulip-terminal#1095]
It's possible to indicate "headings" at different levels, akin to h1-h6 in html, using markdown - this currently just appears as plain text.

This would be a good issue for someone new; here are some suggestions for implementation:
* identify the html generated by the server (inspect in the browser)
* examine the existing tests for `soup2markup` and think how they'd be extended
* add heading styles in themes (and required styles) and add a case to the soup2markup test (this new test should fail)
* explore where the code needs adding in `soup2markup`; add it and ensure the test now passes :)
* test manually in #**test here** on chat.zulip.org :)
* file a PR :)

A later addition would be to add support for multiple levels of heading, such as re-adding the number of leading `#`'s, or some other design.

Please check-in on #**zulip-terminal** on chat.zulip.org if you have questions :)

**Comment by SwarupKharul** [comment]
I would like to work on this issue, please assign it to me!

**Comment by neiljp** [comment]
@SwarupKharul Just calling out like you've done is fine, though you can use `@zulipbot claim` to handle this yourself on Zulip repositories :)

**Comment by SwarupKharul** [comment]
@zulipbot claim

**Comment by zulipbot** [comment]
Welcome to Zulip, @SwarupKharul! We just sent you an invite to collaborate on this repository at https://github.com/zulip/zulip-terminal/invitations. Please accept this invite in order to claim this issue and begin a fun, rewarding experience contributing to Zulip!

Here's some tips to get you off to a good start:
* Join me on the [Zulip developers' server](https://chat.zulip.org), to get help, chat about this issue, and meet the other developers.
* [Unwatch this repository](https://help.github.com/articles/unwatching-repositories/), so that you don't get 100 emails a day.

As you work on this issue, you'll also want to refer to the [Zulip code contribution guide](https://zulip.readthedocs.io/en/latest/contributing/index.html), as well as the rest of the developer documentation on that site.

See you on the other side (that is, the pull request side)!

**Comment by zulipbot** [comment]
Hello @SwarupKharul, you claimed this issue to work on it, but this issue and any referenced pull requests haven't been updated for 10 days. Are you still working on this issue?

If so, please update this issue by leaving a comment on this issue to let me know that you're still working on it. Otherwise, I'll automatically remove you from this issue in 4 days.

If you've decided to work on something else, simply comment `@zulipbot abandon` so that someone else can claim it and continue from where you left off.

Thank you for your valuable contributions to Zulip!

**Comment by avats-dev** [comment]
@kingjuno what about this one?

**Comment by kingjuno** [comment]
@avats-dev I am not working on this much. I guess you can go with this one.

**Comment by mounilKshah** [comment]
@zulipbot claim

**Comment by zulipbot** [comment]
Welcome to Zulip, @mounilKshah! We just sent you an invite to collaborate on this repository at https://github.com/zulip/zulip-terminal/invitations. Please accept this invite in order to claim this issue and begin a fun, rewarding experience contributing to Zulip!

Here's some tips to get you off to a good start:

- Join me on the [Zulip developers' server](https://chat.zulip.org), to get help, chat about this issue, and meet the other developers.
- [Unwatch this repository](https://help.github.com/articles/unwatching-repositories/), so that you don't get 100 emails a day.

As you work on this issue, you'll also want to refer to the [Zulip code contribution guide](https://zulip.readthedocs.io/en/latest/contributing/index.html), as well as the rest of the developer documentation on that site.

See you on the other side (that is, the pull request side)!

**Comment by zulipbot** [comment]
Hello @mounilKshah, you claimed this issue to work on it, but this issue and any referenced pull requests haven't been updated for 10 days. Are you still working on this issue?

If so, please update this issue by leaving a comment on this issue to let me know that you're still working on it. Otherwise, I'll automatically remove you from this issue in 4 days.

If you've decided to work on something else, simply comment `@zulipbot abandon` so that someone else can claim it and continue from where you left off.

Thank you for your valuable contributions to Zulip!

**Comment by mounilKshah** [comment]
Hello,
Sorry for not having updated this issue. 
It is actually my first time working on an open source issue. I have decided to work on the heading-styles of the messages first. Before starting with anything, I went through the conversation on `#feedback` stream on [Zulip developers' server](https://chat.zulip.org/), to understand what is expected.
I have gone through the `developer-feature-tutorial.md` and `developer-file-overview.md` to get an overview of the project. I will try to complete the implementation of various heading styles within this week itself.

**Comment by mounilKshah** [comment]
@zulipbot claim

**Comment by mounilKshah** [comment]
Hello @neiljp 
I tried looking for libraries which would allow the change in font size. However, all I could find was results which said that the only way to change font size in terminal is the user settings, ie, only the user can change the font size.
Hence, I thought of emphasising the heading by changing its colour in the terminal.

**Before `heading-style` implementation:**

![Screenshot 2021-12-27 at 18 26 01](https://user-images.githubusercontent.com/47925568/147513864-51154bea-4c37-4045-89d8-9c81e30b5015.png)

**After `heading-style` implementation:**

![Screenshot 2021-12-27 at 18 26 47](https://user-images.githubusercontent.com/47925568/147513904-2e53b4c2-3e0d-40ad-9376-9dd8f477d713.png)


At present, I have implemented this only in the `zt_dark` theme. Also, all the `heading-styles` have the same implementation at present.

I wanted to ask if these changes are fine and should I continue to implement similar changes in the remaining themes as well or not.
